### PR TITLE
Add test for readline shell

### PR DESCRIPTION
**Pull request by jnoortheen**
internal change: no need for news item.

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### CODE REVIEW

File: tests/test_readline_shell.py [hunk 1]
```diff
@@ -23,3 +23,22 @@ def test_render_completions(prefix, completion, prefix_len, readline_completion)
...
+def test_rl_prompt_cmdloop(line, exp, readline_shell, capsys):
+    shell = readline_shell
+    shell.use_rawinput = False
+    shell.stdin.write(f"{line}\nexit\n")  # note: terminate with '\n'
+    shell.stdin.seek(0)
+    shell.cmdloop()
+    # xonsh, doesn't write all its output to shell.stdout
+    # so capture sys.stdout
+    out, err = capsys.readouterr()
+    assert out.strip() == exp or line
```
Comment (daniel-shimon) [inline tests/test_readline_shell.py:44]: What's `line` doing here? This can never fail

File: tests/test_readline_shell.py [hunk 2]
```diff
@@ -23,3 +23,22 @@ def test_render_completions(prefix, completion, prefix_len, readline_completion)
     assert _render_completions({completion}, prefix, prefix_len) == [
         readline_completion
     ]
+
+
+@pytest.mark.parametrize(
+    "line, exp",
+    [
+        [repr("hello"), None],
```
Comment (daniel-shimon) [inline tests/test_readline_shell.py:31]: Why won't this show hello?

**Review by daniel-shimon — CHANGES_REQUESTED** [review]

File: tests/test_readline_shell.py (hunk 2 above)
Comment (jnoortheen) [reply tests/test_readline_shell.py:31]: it shows hello. instead of repeating, i just use the line again

File: tests/test_readline_shell.py (hunk 1 above)
Comment (jnoortheen) [reply tests/test_readline_shell.py:44]: when exp is none this gets used.

File: tests/test_readline_shell.py (hunk 2 above)
Comment (jnoortheen) [reply tests/test_readline_shell.py]: It is for it to be concise, anyway I just refactored it.

**Comment by jnoortheen** [comment]
@daniel-shimon now you can take a look at this. Now the tests are passing.

**Review by daniel-shimon — APPROVED** [review]
Yeah!

**Pull request merged**